generate.ts

```diff
@@ -406,45 +420,77 @@ function downloadFromUrl(srcUrl: string, appendUrl: string, apiCache, cacheKey:
 }
 
 /////////////////////////////////// Major step 3 - Generate the indicated ouptut files ///////////////////////////////////
-function generateApis(buildIdentifier, targetOutputPathList, buildFlags, apiSrcDescription) {
-    console.log("Generating PlayFab APIs from specs: " + apiSrcDescription);
+interface IGenConfig {
+    branchSpecMap: { [key: string]: string; },
+    delSrc: boolean,
+    buildFlags: string,
+    outputDirs: string[],
+    srcFolder: string,
+    versionKey: string,
+    versionString: string,
+}
+function generateApis(buildIdentifier, target: IBuildTarget) {
+    console.log("Generating PlayFab APIs from specs: " + sdkGeneratorGlobals.apiSrcDescription);
+
+    var genConfig: IGenConfig = null;
+
+    // This is disabled until we more carefully detect and alert on input conflicts
+    //var genConfigPath = path.resolve(target.destPath, "genConfig.json");
+    //try {
+    //    genConfig = require(genConfigPath);
+    //    console.log("Loaded genConfig at: " + genConfigPath);
+    //} catch (_) {
+    //    console.log("Did not find: " + genConfigPath);
+    //}
+
+    if (genConfig) {
+        if (genConfig.buildFlags) target.buildFlags = genConfig.buildFlags.split(" ");
+        if (genConfig.srcFolder) target.srcFolder = genConfig.srcFolder;
+        if (genConfig.versionKey) target.versionKey = genConfig.versionKey;
+        if (genConfig.versionString) target.versionString = genConfig.versionString;
+    }
+
+    if (!target.srcFolder) {
+        throw Error("SdkGenerator/target subfolder not defined: " + target.srcFolder);
```

Comment:
>Error [](start = 14, length = 5)

Will this still display error as we expect it, in Jenkins logs?

Reply:
Changing this from
```throw "string"```
to
```throw Error("string")```
Will significantly improve the stacktrace info that is displayed, and will propagate the failure to jenkins, as expected.